### PR TITLE
net-ftp/atftp: add 0.8.0

### PR DESCRIPTION
**Pull request by madmartin**
Signed-off-by: Martin Dummer <martin.dummer@gmx.net>

### CODE REVIEW

File: net-ftp/atftp/atftp-0.8.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+IUSE="selinux tcpd readline pcre"
+
+DEPEND="tcpd? ( sys-apps/tcp-wrappers )
+	readline? ( sys-libs/readline:0= )
+	pcre? ( dev-libs/libpcre2 )"
+RDEPEND="${DEPEND}
+	!net-ftp/tftp-hpa
+	!net-ftp/uftpd
+	selinux? ( sec-policy/selinux-tftp )"
+BDEPEND="${DEPEND}"
```
Comment (Flowdalic) [inline net-ftp/atftp/atftp-0.8.0.ebuild:23]: Are you sure that all of DEPEND are also BDEPEND?

File: net-ftp/atftp/atftp-0.8.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+PATCHES=(
+	"${FILESDIR}/${P}-CFLAGS.patch"
```
Comment (Flowdalic) [inline net-ftp/atftp/atftp-0.8.0.ebuild:26]: That file is identical the to the already existing `atftp-0.7.5-CFLAGS.patch`. Simply continue using it.
```suggestion
	"${FILESDIR}/${PN}-0.7.5-CFLAGS.patch"
```

File: net-ftp/atftp/atftp-0.8.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+SRC_URI="mirror://sourceforge/atftp/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
+IUSE="selinux tcpd readline pcre"
+
+DEPEND="tcpd? ( sys-apps/tcp-wrappers )
+	readline? ( sys-libs/readline:0= )
+	pcre? ( dev-libs/libpcre2 )"
```
Comment (Flowdalic) [inline net-ftp/atftp/atftp-0.8.0.ebuild:18]:
```suggestion
	pcre? ( dev-libs/libpcre2:= )"
```

**Review by Flowdalic — CHANGES_REQUESTED** [review]
Tests fail for me. Any reason why you dropped `src_test` from the previous version's ebuild? If I re-add it, then the tests run.

**Comment by Flowdalic** [comment]
@madmartin friendly ping

File: net-ftp/atftp/atftp-0.8.0.ebuild (hunk 1 above)
Comment (madmartin) [reply net-ftp/atftp/atftp-0.8.0.ebuild:23]: @Flowdalic IMHO all three of 

```
DEPEND="tcpd? ( sys-apps/tcp-wrappers )
    readline? ( sys-libs/readline:0= )
    pcre? ( dev-libs/libpcre2 )"
```

are needed at build time, yes.
Do you have a different opinion?

File: net-ftp/atftp/atftp-0.8.0.ebuild (hunk 3 above)
Comment (madmartin) [reply net-ftp/atftp/atftp-0.8.0.ebuild:18]: maybe a good idea....

**Comment by madmartin** [comment]
> Tests fail for me. Any reason why you dropped `src_test` from the previous version's ebuild? 

You wrote it in your first sentence. I discovered test.sh fails when used in portage build system/sandbox.

It runs well when started after merge. I did not want to loose time for investigation, so I dropped `src_test()` for now from the ebuild.

> grep: ./atftp-test.CD1J8l/multicast*.log: No such file or directory

is not looking nice but harmless. no logs from multicast test found. I will fix this.

> ./test.sh: line 580: 5 + /data-scratch/var-tmp/portage/net-ftp/atftp-0.8.0/image : syntax error: operand expected (error token is "/data-scratch/var-tmp/portage/net-ftp/atftp-0.8.0/image ")

is more difficult to fix. My first diagnose is that $0 behaves different between regular user shell and portage `src_test()`.

That's why I removed `src_test()` from the ebuild.

File: net-ftp/atftp/atftp-0.8.0.ebuild (hunk 1 above)
Comment (Flowdalic) [reply net-ftp/atftp/atftp-0.8.0.ebuild:23]: Yes they are build dependencies, but I suspect they are not BDEPEND dependencies, just DEPEND dependencies.

**Pull request closed without merging**